torch/csrc/jit/symbolic_script.cpp

```diff
@@ -604,6 +647,18 @@ const std::vector<std::string> functions = {
 
             return output, backward
 
+        def dropout(input,
+                    p: float,
+                    train: bool):
+            mask = torch.empty_like(input)
+            mask.bernoulli_(1 - p)
+            res = mask * input / (1.0 - p)
```

Comment:
Why are we replacing forward code with this? It's not necessary, and avoiding it might open up more optimization opportunities and is less bug prone

Reply:
When I wrote this I had 2 options : 
- changing https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Dropout.cpp#L54 to return the mask as well 
- write a simple dropout implementation in torchscript. 
I went for the latter solution mainly because it's straightforward to do with a few lines. 

I think a design question here is : do we want to keep AD formula as close as possible to its ATen counterpart (for easier possible future deduplication and consistency), or do we want to keep AD only math and simple? In some cases our aten functions considers backend/special sizes etc, I'm not certain if we would like to include those in AD formula or not. Let me know your thoughts on this, would love to have some guidelines on it. Thanks!

Reply:
Fair enough, I forgot that you need the mask! Let's keep it like that.